File: esprit_D2F-formation/src/main/java/esprit/pfe/serviceformation/Services/AnimateurExterneService.java
package esprit.pfe.serviceformation.services;

import esprit.pfe.serviceformation.dto.AnimateurExterneDTO;
import esprit.pfe.serviceformation.dto.AnimateurExterneRequest;
import esprit.pfe.serviceformation.entities.AnimateurExterne;
import org.springframework.data.domain.Page;

import java.util.List;

public interface AnimateurExterneService {
    List<AnimateurExterne> getByBureau(Long bureauId);
    Page<AnimateurExterneDTO> getByBureau(Long bureauId, org.springframework.data.domain.Pageable pageable);
    AnimateurExterne getById(Long bureauId, Long id);
    AnimateurExterne create(Long bureauId, AnimateurExterneRequest request);
    AnimateurExterne update(Long bureauId, Long id, AnimateurExterneRequest request);
    void delete(Long bureauId, Long id);
}

File: esprit_D2F-webapp/src/services/bureau/AnimateurExterneService.ts
import { defaultApi as axios } from "@/services/httpClient";
import { config } from "@/config/env";
import type { AnimateurExterne, AnimateurExterneRequest } from "@/models/bureau";

const bureauUrl = (bureauId: number) =>
  `${config.FORMATION_URL}/formation/bureaux/${bureauId}/animateurs`;

const AnimateurExterneService = {
  async getByBureau(bureauId: number): Promise<AnimateurExterne[]> {
    const response = await axios.get(bureauUrl(bureauId));
    return Array.isArray(response.data) ? (response.data as AnimateurExterne[]) : [];
  },

  async create(bureauId: number, data: AnimateurExterneRequest): Promise<AnimateurExterne> {
    const response = await axios.post(bureauUrl(bureauId), data);
    return response.data;
  },

  async update(bureauId: number, id: number, data: AnimateurExterneRequest): Promise<AnimateurExterne> {
    const response = await axios.put(`${bureauUrl(bureauId)}/${id}`, data);
    return response.data;
  },

  async delete(bureauId: number, id: number): Promise<void> {
    await axios.delete(`${bureauUrl(bureauId)}/${id}`);
  },
};

export default AnimateurExterneService;
